### PR TITLE
VDS-2155: Edit company address

### PR DESCRIPTION
**Pull request by ilyawzrd**
Снимок экрана 2021-08-17 в 12 22 29

Description: 

--

QA-test: 

Demo-test: 

Download artifact URL: ghcr.io/virtocommerce/demo-xapi-app:1.0.0-pr-38-vds-2155-535caace

### CODE REVIEW

**Review by trueboroda — APPROVED** [review]

File: src/app/modules/addresses/addresses.component.ts [hunk 1]
```diff
@@ -63,6 +65,16 @@ export class AddressesComponent implements OnInit, OnDestroy {
...
+  editAddress(address: Address): void {
+    this.store.dispatch(setSelectedAddress({ address }));
+    this.router.navigate([
+      '/addresses',
+      address.id,
+    ]).catch(error => {
+      throw new Error(error);
+    });
+  }
+
```
Comment (asvishnyakov) [inline src/app/modules/addresses/addresses.component.ts:77]: I'd like to use `routerLink` instead and select address by id in component, but not neccessary

File: src/app/modules/addresses/components/address-edit/address-edit.component.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+  ngAfterViewChecked(): void {
+    this.changeDetectorRef.detectChanges();
+  }
+
+  resetForm(): void {
+    this.router.navigate([
+      '/addresses',
+    ]).then(() => {
+      window.location.reload();
```
Comment (asvishnyakov) [inline src/app/modules/addresses/components/address-edit/address-edit.component.ts]: Why?

File: src/app/modules/addresses/store/addresses.effects.ts [hunk 3]
```diff
@@ -32,5 +36,40 @@ export class AddressesEffects {
...
+          command: {
+            memberId: action.id,
+            addresses: [
+              {
+                ...state.editAddress,
+                city: state.editAddress?.city as string,
+                countryCode: state.editAddress?.countryCode as string,
+                countryName: state.editAddress?.countryName as string,
+                line1: state.editAddress?.line1 as string,
+                postalCode: state.editAddress?.postalCode as string,
```
Comment (asvishnyakov) [inline src/app/modules/addresses/store/addresses.effects.ts]: Why? They should pass already in spread operator

File: src/app/modules/addresses/store/addresses.reducer.ts [hunk 4]
```diff
@@ -1,44 +1,71 @@
...
       },
     };
   }),
-  on(AddressesActions.getAddressessFailure, (_): State => ({ addresses: null }))
+  on(AddressesActions.getAddressessFailure, (state): State => ({
+    ...state,
+    addresses: null,
+  }))
 
 );
```
Comment (asvishnyakov) [inline src/app/modules/addresses/store/addresses.reducer.ts:71]: What happens there?

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/app/modules/addresses/store/addresses.effects.ts (hunk 3 above)
Comment (asvishnyakov) [reply src/app/modules/addresses/store/addresses.effects.ts]: Issue with XAPI schema. Won't fix

File: src/app/modules/addresses/store/addresses.reducer.ts (hunk 4 above)
Comment (asvishnyakov) [reply src/app/modules/addresses/store/addresses.reducer.ts:71]: Issue with XAPI schema. Won't fix

File: src/app/modules/addresses/addresses.component.ts (hunk 1 above)
Comment (asvishnyakov) [reply src/app/modules/addresses/addresses.component.ts:77]: Decided left as is

File: src/app/modules/addresses/components/address-edit/address-edit.component.ts (hunk 2 above)
Comment (asvishnyakov) [reply src/app/modules/addresses/components/address-edit/address-edit.component.ts]: Fixed

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**